operations/security_and_authentication/Change_SMNP_credentials_on_Leaf_switches.md

```diff
@@ -0,0 +1,112 @@
+    ```
+
+    Sample output for a system with 2 Dell leaf switches:
+    ```
+    Xname       Brand  Alias
+    x3000c0w14  Dell   sw-leaf-002
+    x3000c0w13  Dell   sw-leaf-001
+    ```
+    
+2. Update SNMP credentials for the `testuser` user on each leaf switch in the system. The SNMP `testuser` user requires 2 password to be provided for the SNMP Authentication and Privacy protocol passwords. Both of these passwords must be 8 characters or longer. In the examples below `foobar01` is the new SNMP Authentication password, and `foobar02` is the new SNMP Privacy password. 
+
+    1.  Aruba leaf switch configuration:
+        ```bash
+        ncn-m001# ssh admin@sw-leaf-001
+        sw-leaf-001# configure terminal
+        sw-leaf-001(config)# snmpv3 user testuser auth md5 auth-pass plaintext foobar01 priv des priv-pass plaintext foobar02
+        sw-leaf-001(config)# exit
+        sw-leaf-001# write memory
+        sw-leaf-001# exit
+        ```
+
+    2.  Dell leaf switch configuration: 
+        ```bash
+        ncn-m001# ssh admin@sw-leaf-001
+        sw-leaf-001# configure terminal
+        sw-leaf-001(config)# snmp-server user testuser cray-reds-group 3 auth md5 foobar01 priv des foobar02
+        sw-leaf-001(config)# exit
+        sw-leaf-001# write memory
+        sw-leaf-001# exit
+        ```
+
+3.  Update Vault with new SNMP credentials:
+    ```bash
+    ncn-m001# SNMP_AUTH_PASS=foobar01
+    ncn-m001# SNMP_PRIV_PASS=foobar02
+    ncn-m001# VAULT_PASSWD=$(kubectl -n vault get secrets cray-vault-unseal-keys -o json | jq -r '.data["vault-root"]' |  base64 -d)
+    ncn-m001# alias vault='kubectl -n vault exec -i cray-vault-0 -c vault -- env VAULT_TOKEN=$VAULT_PASSWD VAULT_ADDR=http://127.0.0.1:8200 VAULT_FORMAT=json vault'
+    ```
+
+    Either update the credentials in Vault for a single leaf switch or update Vault for all leaf switches to have same global default value:
+    -   To update Vault for a single leaf switch:
+        ```bash
+        ncn-m001# XNAME=x3000c0w22
+        ncn-m001# vault kv get secret/hms-creds/$XNAME | 
+            jq --arg SNMP_AUTH_PASS "$SNMP_AUTH_PASS" --arg SNMP_PRIV_PASS "$SNMP_PRIV_PASS" \
+                '.data | .SNMPAuthPass=$SNMP_AUTH_PASS | .SNMPPrivPass=$SNMP_PRIV_PASS' |
+            vault kv put secret/hms-creds/$XNAME -
+        ```
+
+    -   To update Vault for all leaf switches in the system to the same password:
+        ```bash
+        for XNAME in $(cray sls search hardware list --type comptype_mgmt_switch --format json | jq -r .[].Xname); do
+            echo "Updating SNMP creds for $XNAME"
+            vault kv get secret/hms-creds/$XNAME | 
+                jq --arg SNMP_AUTH_PASS "$SNMP_AUTH_PASS" --arg SNMP_PRIV_PASS "$SNMP_PRIV_PASS" \
+                    '.data | .SNMPAuthPass=$SNMP_AUTH_PASS | .SNMPPrivPass=$SNMP_PRIV_PASS' |
+                vault kv put secret/hms-creds/$XNAME -
+        done
+        ``` 
+
+4.  Restart REDS to pickup the new SNMP credentials:
```

Comment:
Need to expand REDS for first time usage in procedure:

Restart the River Endpoint Discovery Service (REDS) to pickup the new SNMP credentials.

Reply:
Should be resolved fixed now.